### PR TITLE
[liblsl] Add port for sccn/liblsl

### PR DESCRIPTION
**Pull request by chausner**
This PR adds a port of the latest release of the [liblsl](https://github.com/sccn/liblsl) (Lab Streaming Layer) library.

### CODE REVIEW

File: ports/liblsl/hardcode-version.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+-		WORKING_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}"
+-		OUTPUT_VARIABLE lslgitbranch
+-		OUTPUT_STRIP_TRAILING_WHITESPACE
+-	)
+-	message(STATUS "Git version information: ${lslgitbranch}/${lslgitrevision}")
+-else()
+-	set(lslgitrevision "unknown")
+-	set(lslgitbranch "unknown")
+-endif()
+ set(LSL_VERSION_INFO "\"git:${lslgitrevision}/branch:${lslgitbranch}/build:${CMAKE_BUILD_TYPE}/compiler:${CMAKE_CXX_COMPILER_ID}-${CMAKE_CXX_COMPILER_VERSION}/boost:\" BOOST_LIB_VERSION")
```
Comment (Codiferous) [inline ports/liblsl/hardcode-version.patch]: @chausner It seems to me that the `lslgitrevision` and `lslgitbranch` variables won't be set after this patch is applied, but we're still using them here. Should we setting them to a hardcoded value here?

**Review by Codiferous — COMMENTED** [review]
When I try to build a test with this library on Windows, I encounter an error that `liblsl32.lib` (on x86-windows) or `liblsl64.lib` (on x64-windows) can't be found. It appears that this port produces `liblsl.lib`, so the name may need to be changed. A super quick search suggests it may be related to [this line in the LSLCMake.cmake file.](https://github.com/sccn/liblsl/blob/d67646467c3d82cbf363492d86b209721d5ae1ff/LSLCMake.cmake#L234)

File: ports/liblsl/hardcode-version.patch (hunk 1 above)
Comment (chausner) [reply ports/liblsl/hardcode-version.patch]: These variables are set in [lines 21-22](https://github.com/Microsoft/vcpkg/pull/5410/files#diff-7a4fca3b946ec4012ba484b7b5fc54adR21) of the portfile.

**Comment by chausner** [comment]
> When I try to build a test with this library on Windows, I encounter an error that liblsl32.lib (on x86-windows) or liblsl64.lib (on x64-windows) can't be found.

Are you encountering the issue when trying to build the library using vcpkg or when building an application that uses the library? Both worked for me when I last tested it. I'm surprised it's looking for a static library as the port currently only supports dynamic linkage.

File: ports/liblsl/hardcode-version.patch (hunk 1 above)
Comment (Codiferous) [reply ports/liblsl/hardcode-version.patch]: Oops! Oversight from me. I see that it's in the portfile -- I looked too quickly at the patch without paying enough attention to the portfile. 😨

**Comment by Codiferous** [comment]
> Are you encountering the issue when trying to build the library using vcpkg or when building an application that uses the library? Both worked for me when I last tested it. I'm surprised it's looking for a static library as the port currently only supports dynamic linkage.

This is happening when building an application that uses the library (I grabbed [this test](https://github.com/sccn/liblsl/blob/master/testing/BounceTest/BounceTest.cpp) and amended the include path in the code). I'm using the `vcpkg integrate install` approach with VS (where it is intended to pick up dependencies automatically).

I tried out the `find_package(LSL CONFIG REQUIRED)` suggestion that prints out after installing and it worked successfully. I definitely see that your PR specifies `LSL_BUILD_STATIC=off` and does `vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)`. I'm not actually even sure what is motivating the project to try to find `liblsl32.lib`.

I don't think this is worth blocking on, as this PR introduces a useful library that does function correctly in the CMake case. Another PR can address the `vcpkg integrate`-related issue, if we can track down the root cause.

**Comment by chausner** [comment]
Thanks!

> I'm not actually even sure what is motivating the project to try to find liblsl32.lib.

liblsl32.lib is specified via a `#pragma` in [lsl_c.h, line 52](https://github.com/sccn/liblsl/blob/d67646467c3d82cbf363492d86b209721d5ae1ff/include/lsl_c.h#L52). I looked into this a bit more and there seem to be two reasons why it currently fails:

1. I am setting `-DLSL_NO_FANCY_LIBNAME=ON` in the portfile which changes the library name to "liblsl.lib" which is incompatible with the `#pragma`. I'll change this back to the default.
2. Due to the different locations and file names of the debug and release variants of liblsl.lib, `LSL_DEBUG_BINDINGS` must be defined in debug configurations for VS to find the library. I haven't found a good solution to this problem yet. For now, I'll print a message in the portfile making users aware of this pitfall.

**Pull request merged**